### PR TITLE
KeyValues update on 6/24/2022

### PR DESCRIPTION
**Pull request by PerfectLaugh**
This fixes linux problem

### CODE REVIEW

**Comment by Wend4r** [comment]
Works on Linux with rebuilt `tier1_i486.a`

**Comment by PerfectLaugh** [comment]
Also have to thanks for @GAMMACASE for his contribution!

**Comment by AnAkkk** [comment]
The vtable is not correct, they just added a destructor at the start of the function:
`virtual ~IKeyValuesSystem() = 0;`

File: public/vstdlib/IKeyValuesSystem.h [hunk 1]
```diff
@@ -23,11 +23,15 @@ typedef int HKeySymbol;
...
 {
 public:
+	virtual ~IKeyValuesSystem() = 0;
+
 	// registers the size of the KeyValues in the specified instance
 	// so it can build a properly sized memory pool for the KeyValues objects
 	// the sizes will usually never differ but this is for versioning safety
 	virtual void RegisterSizeofKeyValues(int size) = 0;
 
+	virtual void Unknown1() = 0;
```
Comment (AnAkkk) [inline public/vstdlib/IKeyValuesSystem.h]: This is still wrong, there is no extra function here.

Comment (PerfectLaugh) [reply public/vstdlib/IKeyValuesSystem.h]: Sorry about that. I don't know compiler mechanism very much

Comment (olokos) [reply public/vstdlib/IKeyValuesSystem.h]: Maybe just replacing it with
`virtual ~IKeyValuesSystem() = 0;`
would work? Just a wild guess.

**Pull request merged**